contracts/GenArt721CoreV3_Engine.sol

```diff
@@ -0,0 +1,2137 @@
+import "@openzeppelin-4.7/contracts/access/Ownable.sol";
+import "./libs/0.8.x/ERC721_PackedHashSeed.sol";
+import "./libs/0.8.x/BytecodeStorage.sol";
+import "./libs/0.8.x/Bytes32Strings.sol";
+
+/**
+ * @title Art Blocks Engine ERC-721 core contract, V3.
+ * @author Art Blocks Inc.
+ * @notice Privileged Roles and Ownership:
+ * This contract is designed to be managed, with progressively limited powers
+ * as a project progresses from active to locked.
+ * Privileged roles and abilities are controlled by the admin ACL contract and
+ * artists. Both of these roles hold extensive power and can arbitrarily
+ * control and modify portions of projects, dependent upon project state. After
+ * a project is locked, important project metadata fields are locked including
+ * the project name, artist name, and script and display details. Edition size
+ * can never be increased.
+ * Care must be taken to ensure that the admin ACL contract and artist
+ * addresses are secure behind a multi-sig or other access control mechanism.
+ * ----------------------------------------------------------------------------
+ * The following functions are restricted to the Admin ACL contract:
+ * - updateArtblocksDependencyRegistryAddress
+ * - updateRenderProviderPrimarySalesAddress
+ * - updatePlatformProviderPrimarySalesAddress
+ * - updateRenderProviderSecondarySalesAddress
+ * - updatePlatformProviderSecondarySalesAddress
+ * - updateRenderProviderPrimarySalesPercentage (up to 100%)
+ * - updatePlatformProviderPrimarySalesPercentage (up to 100%)
+ * - updateRenderProviderSecondarySalesBPS (up to 100%)
+ * - updatePlatformProviderSecondarySalesBPS (up to 100%)
+ * - updateMinterContract
+ * - updateRandomizerAddress
+ * - toggleProjectIsActive
+ * - addProject
+ * - forbidNewProjects (forever forbidding new projects)
+ * - updateDefaultBaseURI (used to initialize new project base URIs)
+ * ----------------------------------------------------------------------------
+ * The following functions are restricted to either the the Artist address or
+ * the Admin ACL contract, only when the project is not locked:
+ * - updateProjectName
+ * - updateProjectArtistName
+ * - updateProjectLicense
+ * - Change project script via addProjectScript, updateProjectScript,
+ *   and removeProjectLastScript
+ * - updateProjectScriptType
+ * - updateProjectAspectRatio
+ * ----------------------------------------------------------------------------
+ * The following functions are restricted to only the Artist address:
+ * - proposeArtistPaymentAddressesAndSplits (Note that this has to be accepted
+ *   by adminAcceptArtistAddressesAndSplits to take effect, which is restricted
+ *   to the Admin ACL contract, or the artist if the core contract owner has
+ *   renounced ownership. Also note that a proposal will be automatically
+ *   accepted if the artist only proposes changed payee percentages without
+ *   modifying any payee addresses, or is only removing payee addresses.)
```

Comment:
This should be updated for the added `autoApproveArtistSplitProposals` logic

Reply:
Great catch–will do so (leaving this comment unresolved until I do so).

Reply:
Addressed here and below.